src/main/java/org/betonquest/betonquest/events/FolderEvent.java

```diff
@@ -87,12 +104,15 @@ public void run() {
                 BetonQuest.event(profile, event);
             }
             if (!chosenList.isEmpty()) {
+                register();
                 new BukkitRunnable() {
                     @Override
                     public void run() {
                         final EventID event = chosenList.remove(0);
-                        if (chosenList.isEmpty()) {
+                        if (event == null || cancelled.remove(profile.getProfileUUID())) {
+                            unregister();
```

Comment:
Registering and unregistering the `EventHandler` on the fly might cause bugs (failure to cancel) if the event is executed for multiple players at once:
1. Event is started for Alice (register is called)
2. Event is started for Bob (register is called)
3. Delay finishes for Alice (unregister is called, inner events are executed)
4. Bob goes offline, Bob's folder event execution is not cancelled as no handler is registered
5. Delay finishes for Bob (unregister is called, inner events are executed)